axum/src/lib.rs

```diff
@@ -337,6 +338,12 @@
 //!     .route_layer(middleware::from_fn(auth));
 //! ```
 //!
+//! The main downside to this approach is that it only works when the async executor being used
+//! has the concept of task-local variables. The example above uses
+//! [tokio's `task_local` macro](https://docs.rs/tokio/1/tokio/macro.task_local.html).
+//! smol does not yet offer equivalent functionality at the time of writing (see
+//! [this GitHub issue](https://github.com/smol-rs/async-executor/issues/139)).
```

Comment:
I'm late, but just to check, axum only supports tokio and can't really run under smol anyway, right? I haven't seen any plans to make it runtime-agnostiv either. Just checking if I'm missing something?

Reply:
No, actually. `axum::serve` uses tokio, but tokio is an optional dependency. There's [`smol-axum`](https://docs.rs/smol-axum/latest/smol_axum/) for an easy way of running an axum service on smol.